Shared UI Components

pinussilvestrus opened the pull request:
<!--

Thanks for creating this pull request!

Please make sure you provide the relevant context.

-->

__Which issue does this PR address?__

Related to #2091 

This makes it easier for other plugins to use the same form components.

__Acceptance Criteria__

<!--

Link the acceptance criteria here if they are defined.

-->

* [ ] Corresponds to the concept <!-- link document here -->
* [ ] Corresponds to the design <!-- link document here -->

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

nikku commented on client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceConfigModal.js:
```diff
@@ -16,7 +16,7 @@ import css from './StartInstanceConfigModal.less';
 import {
   TextInput
-} from '../shared/components';
+} from '../../../app/primitives';
```
Proposal: Find a better name for that folder. Do not move folder under `app` but `shared/sth-sth`. Maybe `shared/ui`?

I'd like to prevent to put more things into the app.

nikku replied:
If these are meant to be shared components, we'd need tests for them, too.

nikku replied:
At least test their existence.

pinussilvestrus replied:
What about other primitives like [modals](https://github.com/camunda/camunda-modeler/tree/develop/client/src/app/primitives/modal), (dropdown) buttons, ...? We also use them outside the scope of the app.

Maybe we should move all of them outside of "primitives" but inside a shared folder called `client/shared/ui`.

barmac replied:
I think they are all UI components, so `shared/ui` sounds like a perfect dir.

nikku replied:
Makes sense.

nikku replied:
Would be a good change, overall. Putting all generic UI (only) building blocks in a single place would benefit the code base I believe.

pinussilvestrus replied:
🚀

pinussilvestrus commented:
Note for the reviewer(s): ba631ea99d1e67cd370d855d36f53d98718f42d3

* moved all general UI components into a folder `shared/ui` (cf. https://github.com/camunda/camunda-modeler/pull/2094#discussion_r573691654)
* added missing test cases for the form UI components

nikku commented on client/src/shared/ui/form/__tests__/RadioSpec.js:
```diff
@@ -0,0 +1,35 @@
+describe('<Radio>', function() {
+
+  it('should render', function() {
+    createRadio;
```
We should execute that one, right?

pinussilvestrus replied:
:o)

pinussilvestrus replied:
Fixed + added another test case on top.

nikku reviewed (changes requested):
One typo, otherwise looking good.

barmac reviewed (approved):
LGTM
